apps/atqr/api/src/app/challenge.controller.ts

```diff
@@ -146,9 +153,40 @@ export class ChallengeController {
   }
 
   // TODO Implement change supervisor endpoint and fix return
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  private changeSupervisor(id: string): Challenge {
-    return {} as Challenge;
+  private async updateSupervisorStatus(
+    id: Guid,
+    updateSupervisorDto: UpdateSupervisorStatusDto
+  ): Promise<void> {
+    const challenge = await this.challengeRepository.findUnique(id);
+    challenge.updateSupervisorStatus(updateSupervisorDto.SupervisorEnum);
+    switch (true) {
```

Comment:
Why are you passing true to the switch statement and making comparisons in the cases?